lib/msf/core/payload/uuid/options.rb

```diff
@@ -111,17 +111,19 @@ def record_payload_uuid(uuid, info={})
   def record_payload_uuid_url(uuid, url)
     return unless datastore['PayloadUUIDTracking']
     # skip if there is no active database
-    return if !(framework.db && framework.db.active)
-
-    payload_info = {
-        uuid: uuid.puid_hex,
-    }
-    payload = framework.db.get_payload(payload_info)
-    unless payload.nil?
-      urls = payload.urls.nil? ? [] : payload.urls
-      urls << url
-      urls.uniq!
-      framework.db.update_payload({id: payload.id, urls: urls})
+    if (framework.db && framework.db.active)
+      payload_info = {
+          uuid: uuid.puid_hex,
+      }
+      payload = framework.db.payloads(payload_info).first
+      unless payload.nil?
+        urls = payload.urls.nil? ? [] : payload.urls
+        urls << url
+        urls.uniq!
+        framework.db.update_payload({id: payload.id, urls: urls})
+      end
+    else
+      print_warning('Without a database connected that payload UUID tracking will not work!')
     end
```

Comment:
I think we need something similar to this here aswell:
https://github.com/rapid7/metasploit-framework/blob/8f729b6fd5ef9ce9483f2bedb123b4f43d8ce764/lib/msf/core/payload/uuid/options.rb#L87

Reply:
@cn-kali-team I think we can get this landed if you're still willing to make this change?

Reply:
@dwelch-r7 Was this change required in the end? 👀

Reply:
@adfoster-r7 nope, I put it in and it ended up just duplicated the message, I didn't realise when I made the comment that if you put it in both places it would do that